src/object.rs

```diff
@@ -725,7 +725,7 @@ macro_rules! glib_object_wrapper {
         // otherwise they would potentially give differeny results for the same object depending on
         // the type we currently know for it
         #[derive(Clone, Hash, Ord)]
-        pub struct $name($crate::object::ObjectRef, ::std::marker::PhantomData<$ffi_name>);
+        pub struct $name($crate::object::ObjectRef);
```

Comment:
That would also apply to `src/shared.rs` and `src/boxed.rs` then and maybe other uses of `PhantomData` here.

I can't remember exactly why this was added (the commit message that added it should have the details) but it was something about variance.

Reply:
It looks like this `PhantomData` dates from https://github.com/gtk-rs/glib/commit/15578ea0fafa53515463f86f104d1d0e5a59f15c.

Looking at the side by side diff, it seems `glib_object_wrapper!` replaced a generic `pub struct GenericObject<T>(Ref, PhantomData<T>);` type.

As I understand, the `PhantomData` no longer serves any purpose now that it's not a generic type, but I suppose the author of that commit didn't consider this at the time the change was made.

Reply:
`glib_shared_wrapper!` and `glib_boxed_wrapper!` don't include `PhantomData` in the the structs they define. `PhantomData` is used by `shared.rs` and `boxed.rs` for a couple generic types, but it's probably actually appropriate there.

So I think this change should be correct, and I don't see other obviously unnecessary uses of `PhantomData`.

Reply:
In shared/boxed that phantom data could be removed by changing the memory manager to a ZST instead of an uninhabitated/uninstantiable type. But doesn't really matter :)

This one can go away though, it's basically the same case as here or not?

https://github.com/gtk-rs/glib/blob/master/src/value.rs#L137

Reply:
Given the comment above that line:

```
// TODO: Should use impl !Send for Value {} once stable
```

Perhaps the `PhantomData` is meant as a hack to prevent `Value` from implementing `Send`? But testing it, it seems it doesn't even with `PhantomData` removed (due to the type of `GValue`) so it shouldn't be needed.

Reply:
Yeah that comment is unrelated. Because `GValue` contains a pointer it won't implement `Send` anyway.